pkg/kubelet/cpumanager/state/state_mem.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/kubelet/cpuset"
+)
+
+type stateMemory struct {
+	assignments   map[string]cpuset.CPUSet
+	defaultCPUSet cpuset.CPUSet
+	pressure      bool
+}
+
+var _ State = &stateMemory{}
+
+// NewMemoryState creates new State for keeping track of cpu/pod assignment
+func NewMemoryState() State {
```

Comment:
This is safe because the only writer to the state is the policy. There is synchronization within the CPU manager to ensure that the policy's callbacks are not executed concurrently.

To improve this, we could either put a big warning on the state saying that it is not thread-safe, or make it thread-safe. Which do you prefer?